### DISCUSSION TITLE
Lost focus in chrome when previousMonth button is disabled

### DISCUSSION THREAD

**Discussion by AlexKDawson** [gpbl/react-day-picker#1676] [Support] [answered]
### Bug description

When using a calendar with a set fromDate, the previous month button will get disabled when you arrive at the earliest possible month. When you navigate to the earliest possible month using this button, chrome will fire a blur/focusout event that unfocuses from the entire calendar (Firefox does not). This poses an issue if you have the calendar in some kind of popper that closes on blur, causing the popper to close when you navigate to the earliest possible month.

### To reproduce

[Sandbox - pardon the event code to exemplify the bug](https://codesandbox.io/s/react-daypicker-forked-l0ji34?file=/src/App.tsx)

#### Steps

1. Open the sandbox in chrome
2. Click the button to show the calendar
3. Click the next month button, to navigate to the next month (the calendar should stay open).
4. Click the last month button, to navigate back to the current and earliest possible month.
5. The last month button will be disabled, chrome will remove focus from the calendar, causing the div to go invisible

### Expected behavior

This is more of a chrome issue, but I was wondering what your thoughts are on doing something to maintain the focus on the calendar. Probably something along the lines of moving the focus before disabling the button.

### Additional context

Whatever you decide, let me know your thoughts. Or if you think theres a workaround that I could apply on the implementation side of things

**Comment by gpbl** [comment]
Interesting bug. Basically, it seems DayPicker is unfocusing its stuff and then Chrome fires a document `focusout` event? I wonder if preventing the propagation of the event could help. Let me give a try!

**Comment by gpbl** [comment, marked as answer]
It looks like the issue is in only Chrome, where the focusout `targetElement` works differently from other browsers. Your app is set to hide the calendar according on focusout, and somehow you enter this bug.

Have you considered to use a "focus trap"? Here is an implementation working with chrome too:

https://codesandbox.io/s/react-day-picker-dialog-iyizc0

**Comment by AlexKDawson** [comment]
I hadn't! Thanks for taking the time to look into this. I think a focus trap would suit the need perfectly here! 

I'm happy enough with that solution that we can close this if you'd like!

**Reply by gpbl** [reply]
No worries, handling focus in the browser is hard stuff :)
